build.gradle

```diff
@@ -204,9 +204,15 @@ task getRevision() {
 			remote = 'origin/main'
 		}
 		def revision = grgit.log(includes:[remote]).size()
-		file('build/revision.txt').text = revision.toString().trim()
+		def revString = revision.toString().trim()
+		def dirty = versioning.info.dirty
+		if (dirty) {
+			// adding the dirty flag to the version
+			revString += "-M"
```

Comment:
Can we instead add a new field to the manifest? That seems like it would be a lot cleaner, and express intent better than `isModified := getAttributes().getValue("Build-Revision").endsWith("-M")`.

Reply:
sure.  I was following the SVN convention, but we can make better choices.